ospfd/ospf_vty.c

```diff
@@ -6330,18 +6785,19 @@ static int show_ip_ospf_database_common(struct vty *vty, struct ospf *ospf,
 		return CMD_WARNING;
 
 	/* `show ip ospf database LSA'. */
-	if (argc == arg_base + 5)
-		show_lsa_detail(vty, ospf, type, NULL, NULL);
+	if ((argc == arg_base + 5) || (uj && (argc == arg_base + 6)))
+		show_lsa_detail(vty, ospf, type, NULL, NULL, json);
```

Comment:
Change 
show_lsa_detail(vty, ospf, type, NULL, NULL, json);
to
show_lsa_detail(vty, ospf, type, NULL, NULL, json_vrf);

Reply:
Done